File: 151002/Rusakovich/NewsDistributedSystem/publisher/src/main/java/by/rusakovich/publisher/controller/NewsController.java
package by.rusakovich.publisher.controller;

import by.rusakovich.publisher.model.dto.news.NewsRequestTO;
import by.rusakovich.publisher.model.dto.news.NewsResponseTO;
import by.rusakovich.publisher.service.impl.NewsService;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping("${api.url}/news")
public class NewsController extends Controller<Long, NewsRequestTO, NewsResponseTO, NewsService> {

    public NewsController(NewsService service) {
        super(service);
    }
}

File: 151004/Bosko/lab5/Publisher/Controllers/V1_0/NewsController.cs
using AutoMapper;
using Microsoft.AspNetCore.Mvc;
using Publisher.Controllers.V1_0.Common;
using Publisher.Entity.Db;
using Publisher.Entity.DTO.RequestTO;
using Publisher.Entity.DTO.ResponseTO;
using Publisher.Service.Interface;

namespace Publisher.Controllers.V1_0
{
    [Route("api/v1.0/news")]
    [ApiController]
    public class NewsController(INewsService NewsService, ILogger<NewsController> Logger, IMapper Mapper)
        : AbstractController<News, NewsRequestTO, NewsResponseTO>(NewsService, Logger, Mapper)
    { }
}
